### ISSUE TITLE
Reusable HTML repr package

### ISSUE THREAD

**Issue by jl-wynen** [scipp/scipp#3305]
I'm writing a HTML repr for `scippneutron.DiskChopper` and can reuse a lot of code of Scipp's repr implementation. But some of it is protected or lacks flexibility.
There will probably be more classes in downstream packages that will get a HTML repr.

We should turn the implementation in Scipp into a reusable package with a proper API so we can use common utilities like preprocessing styles, loading the builtin style(s), icons, etc. This could either be part of Scipp, e.g., `scipp.visualization.html` or a standalone package.

**Comment by YooSunYoung** [comment]
I agree...! Maybe we can have html wrapper per type or we can even use pattern matching here...?

**Comment by jl-wynen** [comment]
Not sure what you mean. We already have separate reprs for each type.

**Comment by jl-wynen** [comment]
How would this differ from `Union[DataGroup, DataArray, Dataset, Variable]`? I.e., what information is present in those that the intermediate repr wouldn't contain. Or what would the intermediate repr contain that is not in those classes?

**Comment by SimonHeybrock** [comment]
The intermediate representation could contain a distilled subset of information, including potential formatting for making it human-readable.
